client/data/site-settings/with-site-settings.js

```diff
@@ -0,0 +1,151 @@
+import { useQueryClient } from '@tanstack/react-query';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { useCallback, useState } from '@wordpress/element';
+import { useTranslate } from 'i18n-calypso';
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+import useSiteSettingsQuery from 'calypso/data/site-settings/use-site-settings-query';
+import useUpdateSiteSettingsMutation from 'calypso/data/site-settings/use-update-site-settings-mutation';
+import { useProtectForm } from 'calypso/lib/protect-form';
+import { recordGoogleEvent, recordTracksEvent } from 'calypso/state/analytics/actions';
+import { errorNotice, successNotice } from 'calypso/state/notices/actions';
+import getCurrentRouteParameterized from 'calypso/state/selectors/get-current-route-parameterized';
+
+const noticeOptions = {
+	duration: 10000,
+	id: 'site-settings-save',
+};
+
+const withSiteSettings = createHigherOrderComponent( ( Wrapped ) => {
+	const WithSiteSettings = ( props ) => {
+		const { siteId } = props;
+		const [ unsavedSettings, setUnsavedSettings ] = useState( {} );
+		const dispatch = useDispatch();
+		const translate = useTranslate();
+		const queryClient = useQueryClient();
+		const path = useSelector( ( state ) => getCurrentRouteParameterized( state, siteId ) );
+		const { markChanged, markSaved } = useProtectForm();
+
+		const trackEvent = useCallback(
+			( name ) => {
+				dispatch( recordGoogleEvent( 'Site Settings', name ) );
+			},
+			[ dispatch ]
+		);
+
+		const trackTracksEvent = useCallback(
+			( name, eventProps ) => {
+				dispatch( recordTracksEvent( name, eventProps ) );
+			},
+			[ dispatch ]
+		);
+
+		const { data, isLoading: isFetchingSettings } = useSiteSettingsQuery( siteId );
+
+		const updateSettings = useCallback(
+			( fields ) => {
+				markChanged();
+				return setUnsavedSettings( { ...unsavedSettings, ...fields } );
+			},
+			[ markChanged, unsavedSettings, setUnsavedSettings ]
+		);
+
+		const {
+			isPending: isSavingSettings,
+			updateSiteSettings: saveSettings,
+			context: mutateContext,
+		} = useUpdateSiteSettingsMutation( siteId, {
+			onMutate: async ( settings ) => {
+				const queryKey = [ 'site-settings', siteId ];
```

Comment:
Query key could be abstracted to a reusable function to ensure we're using the same one.